nano/lib/tomlconfig.hpp

```diff
@@ -0,0 +1,540 @@
+#pragma once
+
+#include <nano/boost/asio.hpp>
+#include <nano/lib/configbase.hpp>
+#include <nano/lib/errors.hpp>
+#include <nano/lib/utility.hpp>
+
+#include <boost/filesystem.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/optional.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+
+#include <cpptoml.h>
+
+namespace nano
+{
+/** Manages a table in a toml configuration table hierarchy */
+class tomlconfig : public nano::configbase
+{
+public:
+	tomlconfig () :
+	tree (cpptoml::make_table ())
```

Comment:
Can the non-template method definitions of this class go into a separate `.cpp` file? (same with `walletconfig.hpp`)

Reply:
Will add a cpp file for walletconfig. I'll keep the tomlconfig as-is for now since it mirrors jsonconfig. I can refactor both in a separate pr.